pkg/services/ngalert/metrics/metrics.go

```diff
@@ -25,17 +26,29 @@ const (
 
 var GlobalMetrics = NewMetrics(prometheus.DefaultRegisterer)
 
+// multi-thread safety and stable ordering.
+type orgRegistries struct {
+	regsMu *sync.Mutex
+	regs   map[int64]prometheus.Registerer
+}
+
+func newOrgRegistries() orgRegistries {
+	return orgRegistries{
+		regsMu: &sync.Mutex{},
+		regs:   make(map[int64]prometheus.Registerer),
+	}
+}
+
 type Metrics struct {
 	*metrics.Alerts
-	AlertState *prometheus.GaugeVec
-	// Registerer is for use by subcomponents which register their own metrics.
-	Registerer           prometheus.Registerer
+	AlertState           *prometheus.GaugeVec
 	RequestDuration      *prometheus.HistogramVec
 	ActiveConfigurations prometheus.Gauge
 	EvalTotal            *prometheus.CounterVec
 	EvalFailures         *prometheus.CounterVec
 	EvalDuration         *prometheus.SummaryVec
 	GroupRules           *prometheus.GaugeVec
+	orgRegistry          orgRegistries
```

Comment:
@gotjosh I'm not sure if this is enough (whether metrics sent correctly) but at least resolves the 
`duplicate metrics collector registration attempted` panic.